libsql/src/hrana/hyper.rs

```diff
@@ -166,3 +184,79 @@ impl crate::statement::Stmt for crate::hrana::Statement<HttpSender> {
         todo!()
     }
 }
+
+#[async_trait::async_trait]
+impl Tx for HttpTransaction<HttpSender> {
+    async fn commit(&mut self) -> crate::Result<()> {
+        self.commit()
+            .await
+            .map_err(|e| crate::Error::Hrana(Box::new(e)))?;
+        Ok(())
+    }
+
+    async fn rollback(&mut self) -> crate::Result<()> {
+        self.rollback()
+            .await
+            .map_err(|e| crate::Error::Hrana(Box::new(e)))?;
+        Ok(())
+    }
+}
+
+#[async_trait::async_trait]
+impl Conn for HttpStream<HttpSender> {
+    async fn execute(&self, sql: &str, params: Params) -> crate::Result<u64> {
+        let mut stmt = Stmt::new(sql, false);
+        bind_params(params, &mut stmt);
+        let result = self
+            .execute(stmt)
+            .await
+            .map_err(|e| crate::Error::Hrana(e.into()))?;
+        Ok(result.affected_row_count)
+    }
+
+    async fn execute_batch(&self, sql: &str) -> crate::Result<()> {
+        let mut batch = Batch::new();
+        let stmts = crate::parser::Statement::parse(sql);
+        for s in stmts {
+            let s = s?;
+            batch.step(None, Stmt::new(s.stmt, false));
+        }
+        self.batch(batch)
+            .await
+            .map_err(|e| crate::Error::Hrana(e.into()))?;
+        Ok(())
+    }
+
+    async fn prepare(&self, sql: &str) -> crate::Result<Statement> {
+        let stmt = crate::hrana::Statement::from_stream(self.clone(), sql.to_string(), true);
+        Ok(Statement {
+            inner: Box::new(stmt),
+        })
+    }
+
+    async fn transaction(
+        &self,
+        _tx_behavior: crate::TransactionBehavior,
+    ) -> crate::Result<crate::transaction::Transaction> {
+        todo!("sounds like nested transactions innit?")
+    }
+
+    fn is_autocommit(&self) -> bool {
+        // TODO: Is this correct?
+        false
+    }
+
+    fn changes(&self) -> u64 {
+        self.affected_row_count()
+    }
+
+    fn last_insert_rowid(&self) -> i64 {
+        self.last_insert_rowid()
+    }
+
+    fn close(&mut self) {
+        //TODO: nobody is using this trait method and since it's not async
+        // there's no reason to do it anyway
+        todo!()
+    }
```

Comment:
We might want to consider removing this if its not needed, we do have drop that we can use to clean things up.

Reply:
Removed. We'd be better to rely on Drop.